src/main/java/com/example/analytics/service/coinMarketCap/CryptoPriceService.java

```diff
@@ -0,0 +1,55 @@
+package com.example.analytics.service.coinMarketCap;
+
+import com.example.analytics.exception.coinMarketCap.ApiException;
+import com.example.analytics.model.coinMarketCap.CryptoCurrencyData;
+import com.example.analytics.repository.coinMarketCap.CryptoCurrencyDataRepository;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+
+import static com.example.analytics.utils.PropertiesReader.getProperty;
+
+@Service
+public class CryptoPriceService {
+    private final String apiUrl = getProperty("crypto.base.url");
```

Comment:
Please, use @Value as a common spring approach to read properties from configuration

Reply:
fixed